src/oInput/oInput.js

```diff
@@ -0,0 +1,98 @@
+import mapEvents from '../utils/eventMapper';
+import './input.css';
+import o from 'ojs-core';
+
+const TYPES = {
+    text: 'Text field',
+    date: 'Date field',
+    email: 'E-mail field',
+    month: 'Month field',
+    number: 'Number field',
+    password: 'Password field',
+    tel: 'Tel field',
+    file: 'File field',
+    url: 'URL field',
+};
+
+class oInput {
+    constructor(config) {
+        this.store = {
+            label: '',
+            type: 'text',
+            typeText: false,
+            placeholder: '',
+            required: true,
+            index: false,
+            labelClass: 'input__label',
+            labelStyle: false,
+            inputClass: 'input__field',
+            inputStyle: false,
+            attributes: [],
+        };
+
+        this.configMerge(config);
+        this.store.events = this.events(config);
+        this.store.id = config.id ? `${config.id}--input` : `${new Date().getTime()}${Math.floor(Math.random() * 100)}--input`;
+    }
+
+    configMerge(config) {
+        Object.assign(this.store, config);
+        if (config.events) {
+            this.store.events = [...config.events, { ...this.events().change }];
+        }
+    }
+
+    events(config) {
```

Comment:
sama metoda mi się wydaję, że mogłaby się nazywać inaczej. Może od czynności, którą robi (chodzi mi o nazwę typu concatEvents na przykład). Pod nazwą events spodziewałbym się bardziej jakiejś tablicy z eventami dla danego inputa na przykład.

Reply:
Taki był początkowy zamysł, że to będzie zwracać tablicę z eventami. Dlatego tam na górze zauważyłeś trochę bubli, które mi umknęły. Ale działa jak ma działać, dlatego ich nie zauważyłem